### PR TITLE
Menu depth fix.

### PR DESCRIPTION
**Pull request by branflake2267**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,3 +22,25 @@ GXT and GWT bindings for Selenium WebDriver.
 
 ##Samples
 * <a href="https://github.com/niloc132/gwt-driver-sample">GWT & GXT Sample</a>
+
+
+##Testing
```
Comment (niloc132) [inline README.md]: lets kick all this out of the homepage's readme into a contrib file or something?

File: src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/BaseTest.java [hunk 2]
```diff
@@ -38,10 +38,11 @@ public void startBrowser() {
 		String url = System.getProperty("webdriver.remote.server");
 		Capabilities c = DesiredCapabilities.chrome();
 		driver = new RemoteWebDriver(c);
-//		driver = new FirefoxDriver();
+		// driver = new FirefoxDriver();
```
Comment (niloc132) [inline src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/BaseTest.java]: ?

File: src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/BaseTest.java [hunk 3]
```diff
@@ -38,10 +38,11 @@ public void startBrowser() {
...
 		driver.manage().timeouts().setScriptTimeout(1000, TimeUnit.MILLISECONDS);
 
-		driver.get("http://"+new NetworkUtils().getNonLoopbackAddressOfThisMachine()+":9080/app/index.html?"+getScenarioName());
+		driver.get("http://" + new NetworkUtils().getNonLoopbackAddressOfThisMachine() + ":9080/app/index.html?"
```
Comment (niloc132) [inline src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/BaseTest.java]: thanks for your opinion

File: src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/MenuBarTest.java [hunk 4]
```diff
@@ -73,4 +74,44 @@ public void testHover() throws Exception {
...
+		// And menu bar has menu
+		Menu fileMenu = menuBar.click("File");
+
+		// When hover over two items (3.0.1 issue), it opens up two menus
+		fileMenu.mouseOver("Open");
+		fileMenu.mouseOver("New");
+
+		// Then a depth of 1 can be found
+		Menu menuFound = GwtWidget.find(Menu.class, driver).atDepth(1).done();
+		Assert.assertNotNull(menuFound);
```
Comment (niloc132) [inline src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/MenuBarTest.java]: can we also assert that the WebElement is the same one as we found when we clicked on 'File'?

File: src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/MenuBarTest.java [hunk 5]
```diff
@@ -73,4 +74,48 @@ public void testHover() throws Exception {
...
+		Menu menuFound1 = GwtWidget.find(Menu.class, driver).atDepth(1).done();
+		Menu menuFound0 = GwtWidget.find(Menu.class, driver).atDepth(0).done();
+		
+		// And those menus have contents of
+		Assert.assertTrue(menuFound1.getElement().getText().contains("Open"));
+		Assert.assertTrue(menuFound0.getElement().getText().contains("Project"));
+	}
+
+	@Test
+	public void testDepthNotFound() throws Exception {
```
Comment (niloc132) [inline src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/MenuBarTest.java]: In maven, doing `mvn clean install -Pintegration-test`, consistently getting a failure from here:

```
-------------------------------------------------------------------------------
Test set: org.senchalabs.gwt.gwtdriver.gxt.models.MenuBarTest
-------------------------------------------------------------------------------
Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 31.016 sec <<< FAILURE!
testDepthNotFound(org.senchalabs.gwt.gwtdriver.gxt.models.MenuBarTest)  Time elapsed: 7.802 sec  <<< FAILURE!
java.lang.AssertionError: There are only 1 menus visible!
    at org.senchalabs.gwt.gwtdriver.gxt.models.Menu$MenuFinder.done(Menu.java:58)
    at org.senchalabs.gwt.gwtdriver.gxt.models.MenuBarTest.testDepthNotFound(MenuBarTest.java:114)
```

Is that test passing for you? As it is an assert, you might need to trap AssertionError as well, but i'm not sure why you aren't getting this.

Comment (branflake2267) [inline src/test/java/org/senchalabs/gwt/gwtdriver/gxt/models/MenuBarTest.java]: Hm, not sure why you're getting it yet. Looking into it. In the IDE it passes. 

screen shot 2013-11-20 at 6 28 06 pm

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,5 +1,32 @@
...
+
+###IDE integration tests
+Building and debugging integration tests
+
+* Run `mvn clean install -Pintegration-test -DskipTests`
+* Run `mvn jetty:run -Pintegration-test`
+* Right click on test and click debug. 
+
+####Assertions
+* Turn on assertions when using Eclipse. Add `-eq` to the integration test VM Arguments.
```
Comment (branflake2267) [inline CONTRIBUTING.md]: ugh

**Pull request merged**